packages/flutter/lib/src/widgets/focus_manager.dart

```diff
@@ -3,50 +3,197 @@
+/// [State.dispose] method, providing the same [FocusNode] to the
+/// [reparentIfNeeded] call each time its [State.build] method is run. In
+/// particular, creating a [FocusNode] each time [State.build] is invoked will
+/// cause the focus to be lost each time the widget is built.
+///
+/// {@template flutter.widgets.focus_manager.focus.keyEvents}
+/// ## Key Event Propagation
+///
+/// The [FocusManager] receives all key events and will pass them to the focused
+/// nodes. It starts with the node with the primary focus, and will call the
+/// [onKey] callback for that node. If the callback returns false, indicating
+/// that it did not handle the event, the [FocusManager] will move to the parent
+/// of that node and call its [onKey]. If that [onKey] returns true, then it
+/// will stop propagating the event. If it reaches the root [FocusScopeNode],
+/// [FocusManager.rootScope], the event is discarded.
+/// {@endtemplate}
+///
+/// ## Focus Traversal
+///
+/// It is often that case that an application will want to give focus to the
+/// logical "next" or "previous" widget in the UI. Calling the [nextFocus],
+/// [previousFocus], and [focusInDirection] functions will perform these
+/// navigation operations on the focus tree.
+///
+/// The policy for what the "next" or "previous" widget is, or the widget in a
+/// particular direction, is determined by the [FocusTraversalPolicy] in force.
+///
+/// The policy that's in force is determined by looking up the widget hierarchy
+/// for a [DefaultFocusTraversal] widget, and obtaining the focus traversal
+/// policy from it. Different focus nodes can have difference policies, so part
+/// of your app can go in widget order, and part can go in reading order,
+/// depending upon the use case.
+///
+/// Predefined policies include [WidgetOrderFocusTraversalPolicy],
+/// [ReadingOrderTraversalPolicy], and [DirectionalFocusTraversalPolicyMixin],
+/// but you can also build your own policies based upon these policies.
 ///
 /// See also:
 ///
-///  * [FocusScopeNode], which is an interior node in the focus tree.
-///  * [FocusScope.of], which provides the [FocusScopeNode] for a given
-///    [BuildContext].
-class FocusNode extends ChangeNotifier {
-  FocusScopeNode _parent;
+///   * [Focus], a widget that manages a [FocusNode] and provides access to
+///     focus information and actions to its descendant widgets.
+///   * [FocusScope], a widget that manages a [FocusScopeNode] and provides
+///     access to scope information and actions to its descendant widgets.
+///   * [FocusManager], a singleton that manages the focus and distributes key
+///     events to focused nodes.
+///   * [FocusTraversalPolicy], an object used to determine how to move the
+///     focus to other nodes.
+///   * [DefaultFocusTraversal], a widget used to configure the default focus
+///     traversal policy for a widget subtree.
+class FocusNode with DiagnosticableTreeMixin, ChangeNotifier {
+  /// Creates a Focus node.
```

Comment:
it doesn't create "a Focus node", it creates "a FocusNode" or "a focus node" or "a node for use in the tree managed by the [FocusManager]" or some such.